drone/src/run.rs

```diff
@@ -66,35 +57,35 @@ async fn drone_main() -> NeverResult {
         .await
         .context("Refreshing certificate.")?;
 
-        futs.push(Box::pin(refresh_loop(cert_options)))
-    }
+        Some(Supervisor::new("cert_refresh", move || {
+            refresh_loop(cert_options.clone())
+        }))
+    } else {
+        None
+    };
 
-    if let Some(proxy_options) = proxy_options {
-        futs.push(Box::pin(serve(proxy_options)));
-    }
+    #[allow(clippy::manual_map)]
+    let _proxy_supervisor = if let Some(proxy_options) = proxy_options {
+        Some(Supervisor::new("proxy", move || {
+            serve(proxy_options.clone())
+        }))
+    } else {
+        None
+    };
 
-    if let Some(agent_options) = agent_options {
-        futs.push(Box::pin(run_agent(agent_options)))
-    }
+    #[allow(clippy::manual_map)]
+    let _agent_supervisor = if let Some(agent_options) = agent_options {
```

Comment:
why not use map here?

Reply:
I can't use map for the ones that require `.clone()`, so I prefer to keep the form consistent for the ones that can.